R-Project/R/fpRerF.R

```diff
@@ -0,0 +1,110 @@
+#' Packs a forest and saves modified forest to disk for use by PackPredict function
+#'
+#' Efficiently packs a forest trained with the RF option.  Two intermediate data structures are written to disk, forestPackTempFile.csv and traversalPackTempFile.csv.  The size of these data structures is proportional to a trained forest and training data respectively.  Both data structures are removed at the end of the operation.  The resulting forest is saved as forest.out.  The size of this file is similar to the size of the trained forest.
+#'
+#' @param X an n by d numeric matrix (preferable) or data frame used to train the forest. 
+#' @param Y a numeric vector of size n.  If the Y vector used to train the forest was not of type numeric then a simple call to as.numeric(Y) will suffice as input.
+#' @param csvFileName the name of a headerless csv file containing combined data and labels.
+#' @param columnWithY is the column in the headerless csv file containing class lables.
+#' @param minParent is the size of nodes that will not be split (minParent=1)
+#' @param numTreesInForest the number of trees to grow in the forest (numTreesInForest=100)
+#' @param numCores is the number of cores to use when training and predicting with the forest (numCores=1)
+#' @param numTreeBins the number of bins to store the forest.  Each bin will contain numTreesInForest/numTreeBins trees.  Only used when forestType=="binned*" (numTreeBins= numCores)
+#' @param forestType the type of forest to grow: rfBase, rerf, inPlace, inPlaceRerF, binnedBase, binnedBaseRerF (forestType="rerf")
+#' @param NodeSizeToBin the minimum node size to use stratified subsampling (NodeSizeToBin=NULL)
+#' @param NodeSizeBin the size of the stratified subsample chosen when NodeSizeToBin criteria is met (NodeSizeBin=NULL)
+#' @param forestType the type of forest to grow: rfBase, rerf, inPlace, inPlaceRerF, binnedBase, binnedBaseRerF (forestType="rerf")
+#' @param mtry the number of features to consider when splitting a node (mtry=ncol(X)^.5)
+#' @param mtryMult the average number of features combined to form a new feature when using RerF (mtryMult=1)
+#'
+#'
+#' @export
+#'
+#' @examples
+#' library(rerf)
+#' forest <- fpRerF(as.matrix(iris[, 1:4]), as.numeric(iris[[5L]])-1)
+#'
+fpRerF <-
+	function(X=NULL, Y=NULL,csvFileName=NULL, columnWithY=NULL,minParent=1, numTreesInForest=100, numCores=1,numTreeBins=NULL, forestType="rerf", nodeSizeToBin=NULL, nodeSizeBin=NULL,mtry=NULL, mtryMult=NULL,seed=sample(1:1000000,1)){
+
+		##### Basic Checks
+		################################################
+		if(numCores < 1){
+			stop("at least one core must be used.")
+		}
+		if(minParent < 1){
+			stop("at least one observation must be used in each node.")
+		}
+		if(numTreesInForest < 1){
+			stop("at least one tree must be used.")
+		}
+		if(is.null(forestType)){
```

Comment:
This would only occur if someone passed NULL in for the forestType. Seems not needed.